app/main.py

```diff
@@ -1,2 +1,77 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self.capacity = 8
+        self.load_factor = 2 / 3
+        self.size = 0
+        self.hash_table = [None for _ in range(self.capacity)]
+
+    def _size_hash_table(self):
+        self.capacity *= 2
+        self.size = 0
+        new_hash_table = [elem for elem in self.hash_table if elem]
+        self.hash_table = [None for _ in range(self.capacity)]
+        for elem in new_hash_table:
+            if elem:
+                self.add_hash_table(elem[0], elem[1])
+
+    def add_hash_table(self, key, value):
+        hash_key = hash(key)
+        index_ = hash_key % self.capacity
+
+        if self.hash_table[index_] is None or \
+                self.hash_table[index_][0] == key:
+            self.hash_table[index_] = (key, value, hash_key)
+            self.size += 1
+        else:
+            while True:
+                if index_ == self.capacity:
+                    index_ = 0
+
+                if self.hash_table[index_] is None:
+                    self.hash_table[index_] = (key, value, hash_key)
+                    self.size += 1
+                    break
+
+                if self.hash_table[index_][0] == key:
+                    self.hash_table[index_] = (key, value, hash_key)
+                    self.size += 1
+                    break
+                index_ += 1
```

Comment:
It can be much more concise. It is relevant to use such incrementation: `index_ = (index_ + 1) % self.capacity`.